service/create/service_test.go

```diff
@@ -0,0 +1,239 @@
+package create
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+func TestAllExistingInstancesMatch(t *testing.T) {
+	tests := []struct {
+		name      string
+		instances *ec2.DescribeInstancesOutput
+		state     EC2StateCode
+		res       bool
+	}{
+		{
+			name: "Expect terminated with a single terminated instance in a single reservation",
+			instances: &ec2.DescribeInstancesOutput{
+				Reservations: []*ec2.Reservation{
+					{
+						Instances: []*ec2.Instance{
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2TerminatedState)),
+								},
+							},
+						},
+					},
+				},
+			},
+			state: EC2TerminatedState,
+			res:   true,
+		},
+		{
+			name: "Expect terminated with three terminated instances in a single reservation",
+			instances: &ec2.DescribeInstancesOutput{
+				Reservations: []*ec2.Reservation{
+					{
+						Instances: []*ec2.Instance{
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2TerminatedState)),
+								},
+							},
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2TerminatedState)),
+								},
+							},
+							{
+								State: &ec2.InstanceState{
+									Code: aws.Int64(int64(EC2TerminatedState)),
+								},
+							},
+						},
+					},
+				},
+			},
+			state: EC2TerminatedState,
+			res:   true,
+		},
+		{
+			name: "Expect not terminated a terminated instance and a running instance in a single reservation",
```

Comment:
`Expect not terminated a terminated` -> `Expect not terminated with a terminated`.